src/test/java/com/snowflake/kafka/connector/internal/streaming/TopicPartitionChannelTest.java

```diff
@@ -591,7 +598,7 @@ public void testInsertRows_ValidationResponseHasErrors_ErrorTolerance_ALL_LogEna
         topicPartitionChannel.new StreamingBuffer();
     streamingBuffer.insert(records.get(0));
 
-    assert topicPartitionChannel.insertBufferedRecords(streamingBuffer).hasErrors();
+    assert topicPartitionChannel.insertBufferedRecordsWithRetry(streamingBuffer, 1).hasErrors();
```

Comment:
consider adding tests with maxRetry count >1
i think that is where the new code will kick in right?